instrumentation/avaje-jex-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/avaje/jex/v3_0/JexTest.java

```diff
@@ -0,0 +1,137 @@
+    String content = response.contentUtf8();
+
+    assertThat(content).isEqualTo(id);
+    assertThat(response.status().code()).isEqualTo(200);
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET /param/{id}")
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(URL_SCHEME, "http"),
+                            equalTo(URL_PATH, "/param/" + id),
+                            equalTo(HTTP_REQUEST_METHOD, "GET"),
+                            equalTo(HTTP_RESPONSE_STATUS_CODE, 200),
+                            satisfies(USER_AGENT_ORIGINAL, val -> val.isInstanceOf(String.class)),
+                            equalTo(HTTP_ROUTE, "/param/{id}"),
+                            equalTo(NETWORK_PROTOCOL_VERSION, "1.1"),
+                            equalTo(SERVER_ADDRESS, "localhost"),
+                            equalTo(SERVER_PORT, port),
+                            equalTo(CLIENT_ADDRESS, "127.0.0.1"),
+                            equalTo(NETWORK_PEER_ADDRESS, "127.0.0.1"),
+                            satisfies(NETWORK_PEER_PORT, val -> val.isInstanceOf(Long.class)))));
+  }
+
+  @Test
+  void testSpanNameAndHttpRouteSpanResponseError() {
+    client.get("/error").aggregate().join();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET /error")
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(URL_SCHEME, "http"),
+                            equalTo(URL_PATH, "/error"),
+                            equalTo(HTTP_REQUEST_METHOD, "GET"),
+                            equalTo(HTTP_RESPONSE_STATUS_CODE, 500),
+                            satisfies(USER_AGENT_ORIGINAL, val -> val.isInstanceOf(String.class)),
+                            equalTo(HTTP_ROUTE, "/error"),
+                            equalTo(NETWORK_PROTOCOL_VERSION, "1.1"),
+                            equalTo(SERVER_ADDRESS, "localhost"),
+                            equalTo(SERVER_PORT, port),
+                            equalTo(ERROR_TYPE, "500"),
+                            equalTo(CLIENT_ADDRESS, "127.0.0.1"),
+                            equalTo(NETWORK_PEER_ADDRESS, "127.0.0.1"),
+                            satisfies(NETWORK_PEER_PORT, val -> val.isInstanceOf(Long.class)))));
+  }
+
+  @Test
+  void testHttpRouteMetricWithPathParamResponseSuccessful() {
+    String id = "123";
+    AggregatedHttpResponse response = client.get("/param/" + id).aggregate().join();
+    String content = response.contentUtf8();
+    String instrumentation = "io.opentelemetry.jetty-12.0";
```

Comment:
adding a note because it confused me as to why we are testing the jetty instrumentation here: this new instrumentation does not generate any spans or metrics on its own, but rather it will just update any existing SERVER span context with this route information. And therefore we use jetty to validate that the `HTTP_ROUTE` was added to the context and subsequently included in span name/tags and metrics tags.

Reply:
Indeed